lib/migration_helper.rb

```diff
@@ -13,6 +13,40 @@ def sanitize_sql_for_conditions(conditions)
   end
 
   #
+  # Batching
+  #
+
+  def say_batch_started(count)
+    say "Processing #{count} rows", :subitem
+    @batch_total_started = Time.now.utc
+    @batch_started = Time.now.utc
+    @batch_total = count
+    @batch_count = 0
+  end
+
+  def say_batch_processed(count)
+    Thread.exclusive do
+      @batch_count += count
+
+      progress = @batch_count / @batch_total.to_f * 100
+      timing   = Time.now.utc - @batch_started
+      estimate = estimate_batch_complete(@batch_total_started, progress)
+
+      say "#{count} rows (#{"%.2f" % progress}% - #{@batch_count} total - #{"%.2f" % timing}s - ETA: #{estimate})", :subitem if count > 0
```

Comment:
Good catch...don't think there can be a div by zero, but even so it's not required to calculate any of that since I won't be showing it.